Extension/lib/filter/subscription.js

```diff
@@ -108,6 +108,99 @@ adguard.subscriptions = (function (adguard) {
     };
 
     /**
+     * Parses filter metadata from rules header
+     *
+     * @param rules
+     * @returns object
+     */
+    var parseFilterDataFromHeader = function (rules) {
+        function parseTag(tagName) {
+            var result = '';
+
+            //Look up only 50 first lines
+            for (var i = 0; i < 50; i++) {
+                var r = rules[i];
+
+                var search = '! ' + tagName + ': ';
+                var indexOf = r.indexOf(search);
+                if (indexOf >= 0) {
+                    result = r.substring(indexOf + search.length);
+                }
+            }
+
+            return result;
+        }
+
+        return {
+            name: parseTag('Title'),
+            description: parseTag('Description'),
+            homepage: parseTag('Homepage'),
+            version: parseTag('Version'),
+            timeUpdated: parseTag('TimeUpdated'),
+            expires: parseTag('Expires')
+        }
+    };
+
+    var addFilterId = function () {
+        var max = 0;
+        filters.forEach(function (f) {
+            if (f.filterId > max) {
+                max = f.filterId;
+            }
+        });
+
+        return max >= 1000 ? max + 1 : 1000;
+    };
+
+    /**
+     * Adds or updates custom filter
+     *
+     * @param url subscriptionUrl
+     * @param callback
+     */
+    var updateCustomFilter = function (url, callback) {
+
+        adguard.backend.loadFilterRulesBySubscriptionUrl(url, function (rules) {
+            //Check if filter from this url was added before
+            var filter = filters.find(function (f) {
```

Comment:
You don't check that the new version differs from the previous. So after updating it's always showing popup about updating.